### PR TITLE
[Feature][Fix](multi-catalog) Implements transactional hive full acid tables.

### PR DESCRIPTION
**Pull request by kaka11chen**
## Proposed changes

After supporting insert-only transactional hive full acid tables https://github.com/apache/doris/pull/19518, https://github.com/apache/doris/pull/19419, this PR support transactional hive full acid tables.

- Support hive3 transactional hive full acid tables.
- Hive2 transactional hive full acid tables need to run major compactions.

## Further comments
Regression test only test with hive2 transactional hive full acid tables now, because the docker image version is hive 2.x, will add hive 3.x. docker later.

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java [hunk 1]
```diff
@@ -114,17 +114,19 @@ public static String printBroker(String brokerName, TNetworkAddress address) {
...
             String filePath,
             List<String> columnsFromPath,
-            boolean caseSensitive)
+            boolean caseSensitive,
+            boolean isACID)
             throws UserException {
         if (columnsFromPath == null || columnsFromPath.isEmpty()) {
             return Collections.emptyList();
         }
+        int pathCount = isACID ? 3 : 2;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java:129]: Add comment explain `3` and `2`, better give an example

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/HiveScanNode.java [hunk 2]
```diff
@@ -192,12 +195,50 @@ private void getFileSplitByPartitions(HiveMetaStoreCache cache, List<HivePartiti
...
                     allFiles.addAll(splitFile(status.getPath(), status.getBlockSize(),
                             status.getBlockLocations(), status.getLength(), status.getModificationTime(),
-                            isSplittable, fileCacheValue.getPartitionValues()));
+                            isSplittable, fileCacheValue.getPartitionValues(), fileCacheValue.getAcidInfo()));
                 }
             }
         }
     }
 
+    private List<Split> splitFile(Path path, long blockSize, BlockLocation[] blockLocations, long length,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/HiveScanNode.java:204]: Better to find a way to extract the common part of `splitFile` and the same method defined in `FileScanNode`, or it is hard to maintain.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java (hunk 1 above)
Comment (kaka11chen) [reply fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java:130]: ok

**Review by morningman — APPROVED** [review]
LGTM

**Comment by kaka11chen** [comment]
run buildall

**Pull request merged**